s2a.go

```diff
@@ -344,3 +349,63 @@ func getVerificationMode(verificationMode VerificationModeType) s2av2pb.Validate
 		return s2av2pb.ValidatePeerCertificateChainReq_UNSPECIFIED
 	}
 }
+
+// NewS2ADialTLSContextFunc returns a dial func which establishes an MTLS connection using S2A.
+// example use with http.RoundTripper:
+//
+//		dialTLSContext := s2a.NewS2aDialTLSContextFunc(&s2a.ClientOptions{
+//			S2AAddress:         s2aAddress, // required
+//			EnableV2:           true, // must be true
+//		})
+//	 	trans := http.DefaultTransport
+//	 	trans.DialTLSContext = dialTLSContext
+func NewS2ADialTLSContextFunc(opts *ClientOptions) func(ctx context.Context, network, addr string) (net.Conn, error) {
+
+	return func(ctx context.Context, network, addr string) (net.Conn, error) {
+
+		fallback := func(err error) (net.Conn, error) {
+			if opts.FallbackOpts != nil && opts.FallbackOpts.FallbackDialer != nil &&
+				opts.FallbackOpts.FallbackDialer.Dialer != nil && opts.FallbackOpts.FallbackDialer.ServerAddr != "" {
+				fbDialer := opts.FallbackOpts.FallbackDialer
+				grpclog.Infof("fall back to dial: %s", fbDialer.ServerAddr)
+				fbConn, fbErr := fbDialer.Dialer.DialContext(ctx, network, fbDialer.ServerAddr)
```

Comment:
micro-nit: s/fbErr/err

Reply:
in this case, there is an `err` from S2A failure, that's passed in to the func, and then `fbErr` is for the fallback dial, we need to differentiate the two.